client/src/routes/dashboard.svelte

```diff
@@ -0,0 +1,93 @@
+<script context="module" lang="ts">
+	import type { Load } from '@sveltejs/kit';
+	export const load: Load = async ({ session }) => {
+		if (!session?.user) {
+			return {
+				status: 302,
+				redirect: '/login',
+			};
+		}
+		const res = await getExpenses({ userId: session.user.id });
+		return {
+			status: res.status,
+			props: {
+				expenses: res.data,
+			},
+		};
+	};
+</script>
+
+<script lang="ts">
+	import { Expense } from '$mock/api/models/expense_model';
+
+	import Drawer, { AppContent, Content } from '@smui/drawer';
+	import List, { Item, Text } from '@smui/list';
+	import { getExpenses } from '$api/dashboard';
+
+	export let expenses: Expense[] = [];
+	let clickedExpense: Expense | null = null;
+	// is this allowed? im not sure
+	if (clickedExpense === null && expenses) {
+		clickedExpense = expenses[0];
+	}
+
+	function onDrawerClick(expense: Expense) {
+		clickedExpense = expense;
+	}
+</script>
+
+<div class="drawer-container">
+	<Drawer>
+		<Content>
+			<List>
+				{#each expenses as expense, i (i)}
+					<Item href="javascript:void(0)" on:click={() => onDrawerClick(expense)}>
```

Comment:
Using _href="javascript:void(0)"_ to prevent the default behavior of clicking?

Usually there is a way in frameworks to "preventDefault" (and also to "stopPropogation").

According to: https://svelte.dev/tutorial/event-modifiers
You can use `on:click|preventDefault` instead of `on:click`, and delete href attribute.
Hopefully it works :D Did not try it .

Reply:
That was just some sloppy copy-paste, I am not sure why that was there. It does not make a difference if I remove it, preventDefault isn't needed either.